cpp/src/io/avro/reader_impl.cu

```diff
@@ -138,24 +146,26 @@ class metadata : public file_metadata {
   datasource* const source;
 };
 
-rmm::device_buffer reader::impl::decompress_data(const rmm::device_buffer& comp_block_data,
-                                                 rmm::cuda_stream_view stream)
+rmm::device_buffer decompress_data(datasource* source,
+                                   metadata* meta,
```

Comment:
If `source` and `meta` are read only, please add const qualifier for the pointers.
```suggestion
rmm::device_buffer decompress_data(datasource const* source,
                                   metadata const* meta,
```

Reply:
Why use raw pointers at all?

Reply:
`const&` is a better fit here. I used pointers because it was the least amount of alteration required to remove the reader and impl classes, which is the focus of the PR. I should have checked the code over after that goal was accomplished to make sure the result of the refactor still made sense.

Reply:
unfortunately metadata and datasource cannot be const.